packages/drivers/odsp-driver/src/odspDelayLoadedDeltaStream.ts

```diff
@@ -238,6 +238,8 @@ export class OdspDelayLoadedDeltaStream {
 
 		await new Promise<void>((resolve, reject) => {
 			this.joinSessionRefreshTimer = setTimeout(() => {
+				this.clearJoinSessionTimer();
+				// Clear the timer as it is going to be scheduled again as part of refreshing join session.
```

Comment:
This is adding a call to `clearJoinSessionTimer()` before calling `joinSession`. `[joinSession` already does this](https://github.com/microsoft/FluidFramework/pull/18837/files#diff-f08c3c52f6e4ff00dd2446fa73e8643750f3fb96be84c008ef389d4322383616R273), to ensure the timer is never scheduled twice.

The goal here would be to figure out why we call `joinSession` twice, once when the first timer is still active. If that's expected or if it's not a huge concern (would be very expensive and risky to add coordination/locking here), we can just remove the error event and just clear the timer if exists, when `joinSession` is called.

Reply:
No, the link that you pasted is an error case. We don't go into that if condition usually. So, the logic before was that when the timer when off, we call this.joinSession() again but the timer is never set to undefined before calling joinSession. So, clearing the timer does that. Previously even in normal flow, that event was always getting recorded because of this.

Reply:
So, the test that I added was failing before as it was recording that event.

Reply:
Oh, I think I see. So, we're keeping the error, but the issue is that the error is happening way too often in the regular flow, hence the noise it's creating.